src/script-handlers/slotHandler.ts

```diff
@@ -48,6 +55,57 @@ export default function eventHandler(documentation: Documentation, path: NodePat
         }
         this.traverse(pathMember)
       },
+      visitJSXElement(pathJSX: NodePath<bt.JSXElement>) {
+        const tagName = pathJSX.node.openingElement.name
+        if (bt.isJSXIdentifier(tagName) && tagName.name === 'slot') {
+          const doc = documentation.getSlotDescriptor(getName(pathJSX))
+          doc.description = getDescription(pathJSX)
+        }
+        this.traverse(pathJSX)
+      },
     })
   }
 }
+
+function getName(pathJSX: NodePath<bt.JSXElement>): string {
+  const oe = pathJSX.node.openingElement
+  const names = oe.attributes.filter(
+    (a: bt.JSXAttribute) => bt.isJSXAttribute(a) && a.name.name === 'name',
+  ) as bt.JSXAttribute[]
+
+  const nameNode = names.length ? names[0].value : null
+  return nameNode && bt.isStringLiteral(nameNode) ? nameNode.value : 'default'
+}
+
+function getDescription(pathJSX: NodePath<bt.JSXElement>): string {
+  let indexInParent = -1
+  const siblings = (pathJSX.parentPath.node as bt.JSXElement).children
+  if (!siblings) {
+    return ''
+  }
+  siblings.forEach((node: bt.JSXElement, i: number) => {
+    if (node === pathJSX.node) {
+      indexInParent = i
+    }
+  })
+
+  let commentExpression: bt.JSXExpressionContainer | null = null
+  do {
+    const currentNode = siblings[indexInParent]
+    if (bt.isJSXExpressionContainer(currentNode)) {
+      commentExpression = currentNode
+    }
+  } while (!commentExpression && indexInParent--)
```

Comment:
Doesn't this loop forever is there is no preceding JSXExpressionContainer? `&& indexInParent >= 0`. Also I think I'd prefer a `for` loop that ascends and `break`s when hitting the target, it makes it clearer when the decrement happens and what the looping conditions are.

Reply:
`indexInParent--` is a multicrazy usage thingy that will return `0` which is like `false` in js.
It is ugly I have to agree. I will have it a for before it merges in.